### PR TITLE
Mes 7805 cat b unsuccessful serious serious

### PR DESCRIPTION
**Pull request by solomonsimukoko**
Mes 7805 cat b unsuccessful serious serious

## Checklist

- [ ] PR title includes the JIRA ticket number and an understandable description
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket

### CODE REVIEW

File: src/element-references/des-element-references.json [hunk 1]
```diff
@@ -257,6 +258,7 @@
     },
   "des-debrief-screen":{
       "test-outcome-container" : ".test-outcome-container",
+      "status-fail" : "#status-fail",
```
Comment (turnerc1) [inline src/element-references/des-element-references.json:261]: This already exists as "test-outcome-failed", 3 lines below

File: src/features/des/cat-b/UnsuccessfulTests/UnsuccessfulTestWithSeriousFaults.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
+Feature: Cat-B Unsuccessful with serious faults feature
+  Scenario:   Successful test with 15 faults
```
Comment (turnerc1) [inline src/features/des/cat-b/UnsuccessfulTests/UnsuccessfulTestWithSeriousFaults.feature:2]: This is not a tests with 15 faults, it is a tests with serious faults. Change the name please.

File: src/features/des/cat-b/UnsuccessfulTests/UnsuccessfulTestWithSeriousFaults.feature [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      | showMeQuestion          | S1 - Rear windscreen                        |
+      | weatherCondition        | 1 - Bright / dry roads                      |
+      | seriousComment          | Not done well                               |
+    And I click on the button "des-office-page::mark-as-complete"
+    And I wait on element "des-office-page::mark-as-complete-modal-title" to be displayed
+    And I click on the button "des-office-page::mark-as-complete-modal-submit-btn"
+    And I expect that element "des-my-journal::journal-loading-spinner" is not displayed
+    And I wait on element "#activity-code-20654334" to be displayed
+    And I expect that element "#activity-code-20654334" matches the text "2"
+    And I wait on element "//ion-header//ion-buttons/ion-back-button" to be displayed
```
Comment (matthew2564) [inline src/features/des/cat-b/UnsuccessfulTests/UnsuccessfulTestWithSeriousFaults.feature:52]: Should this be using the global elements alias pattern?

File: src/pageobjects/des/nonPassFinalisation.pageobject.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  get passFinalisationContinueButton() { return ('des-final-outcome-screen::continue-btn'); }
+
+  checkTestOutcomeVisibility(testOutcomeText:string) {
+    $(`#test-outcome-text=='${testOutcomeText}'`).waitForDisplayed();
+  }
+
+  selectTerminationReasonFromList(terminationReason:string) {
+    $('ion-alert').waitForDisplayed();
+    $(`.alert-radio-label=${terminationReason}`).click();
+  }
```
Comment (matthew2564) [inline src/pageobjects/des/nonPassFinalisation.pageobject.ts:45]: Are these used? If so, they should be async/await.

File: src/features/des/cat-b/UnsuccessfulTests/UnsuccessfulTestWithSeriousFaults.feature [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    And I wait on element "des-debrief-screen::status-fail" to be displayed
+    Then I expect that element ".counter-label" matches the text "Control - Accelerator"
+    And I click on the button "des-debrief-screen::end-debrief-btn"
+    And I click on the button "des-general::continue-to-non-pass-finalisation"
+    And I complete the unsuccessful final outcome screen from a data table
+      | d255                          | na              |
+      | testLanguage                  | na              |
+      | debriefWitnessed              | yes             |
+    And I pause for 2000ms
+    And I wait on element "des-final-confirmation-screen::page-title" to be displayed
```
Comment (turnerc1) [inline src/features/des/cat-b/UnsuccessfulTests/UnsuccessfulTestWithSeriousFaults.feature:30]: Add the the step to check for the confirmation screen, check the other feature files for it. It should be, something like "And I check the confirmation page details from a data table"

File: src/features/des/cat-b/UnsuccessfulTests/UnsuccessfulTestWithSeriousFaults.feature [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    And I click on the button "des-test-report-screen::end-test-continue-to-debrief-btn"
+    And I wait on element "des-debrief-screen::status-fail" to be displayed
+    Then I expect that element ".counter-label" matches the text "Control - Accelerator"
+    And I click on the button "des-debrief-screen::end-debrief-btn"
+    And I click on the button "des-general::continue-to-non-pass-finalisation"
+    And I complete the unsuccessful final outcome screen from a data table
+      | d255                          | na              |
+      | testLanguage                  | na              |
+      | debriefWitnessed              | yes             |
+    And I pause for 2000ms
```
Comment (turnerc1) [inline src/features/des/cat-b/UnsuccessfulTests/UnsuccessfulTestWithSeriousFaults.feature:29]: Is this pause needed?

**Review by turnerc1 — CHANGES_REQUESTED** [review]

File: src/pageobjects/des/nonPassFinalisation.pageobject.ts [hunk 7]
```diff
@@ -0,0 +1,133 @@
+import checkEqualsText from '@shared-boilerplate/support/check/checkEqualsText';
```
Comment (turnerc1) [inline src/pageobjects/des/nonPassFinalisation.pageobject.ts:1]: I think that these can just go in the finalise outcome page. The non-pass and pass finalisation page are the same.

**Review by turnerc1 — CHANGES_REQUESTED** [review]

**Review by gabrielngdvsa — APPROVED** [review]

File: src/element-references/des-element-references.json [hunk 8]
```diff
@@ -8,7 +8,8 @@
     "candidate-driver-number": "#candidate-driver-number",
     "signature-pad-area": "#signature-pad",
     "retry-signature-btn": "#retry-icon",
-    "page-title": "#page-title"
+    "page-title": "#page-title",
+    "continue-to-non-pass-finalisation": "#continue-to-non-pass-finalisation"
```
Comment (turnerc1) [inline src/element-references/des-element-references.json:12]: is this in the correct place?

File: src/element-references/des-element-references.json [hunk 9]
```diff
@@ -400,6 +401,7 @@
       "mark-as-complete-modal-title": "#model-alert-title-text",
       "mark-as-complete-modal-submit-btn":"//*[@id='continue-button'][contains(text(), 'Continue')]",
       "fault-comment-input": "#fault-comment-text-driving-controlsAccelerator",
-      "fault-comment-eyesight-input": "#fault-comment-text-serious-eyesightTest"
+      "fault-comment-eyesight-input": "#fault-comment-text-serious-eyesightTest",
+      "serious-comment": "#fault-comment-text-serious-controlsAccelerator"
```
Comment (turnerc1) [inline src/element-references/des-element-references.json:405]: change to represent what fault it is for please. e.g. "serious-comment-controls-accelerator"

File: src/element-references/des-element-references.json [hunk 10]
```diff
@@ -400,6 +401,7 @@
       "mark-as-complete-modal-title": "#model-alert-title-text",
       "mark-as-complete-modal-submit-btn":"//*[@id='continue-button'][contains(text(), 'Continue')]",
       "fault-comment-input": "#fault-comment-text-driving-controlsAccelerator",
-      "fault-comment-eyesight-input": "#fault-comment-text-serious-eyesightTest"
+      "fault-comment-eyesight-input": "#fault-comment-text-serious-eyesightTest",
```
Comment (turnerc1) [inline src/element-references/des-element-references.json:404]: change to represent what fault type it is for please. e.g. "serious-fault-comment-eyesight-input"

File: src/element-references/des-element-references.json [hunk 11]
```diff
@@ -400,6 +401,7 @@
       "mark-as-complete-modal-title": "#model-alert-title-text",
       "mark-as-complete-modal-submit-btn":"//*[@id='continue-button'][contains(text(), 'Continue')]",
       "fault-comment-input": "#fault-comment-text-driving-controlsAccelerator",
```
Comment (turnerc1) [inline src/element-references/des-element-references.json:403]: change to represent what fault it is for please. e.g. "driving-fault-controls-acelerator-comment-input"

**Review by turnerc1 — CHANGES_REQUESTED** [review]

File: src/element-references/des-element-references.json (hunk 8 above)
Comment (Cullima) [reply src/element-references/des-element-references.json:12]: agree should belong to the page

**Review by matthew2564 — APPROVED** [review]

**Pull request merged**